src/control/security/grpc_authorization_test.go

```diff
@@ -0,0 +1,92 @@
+//
+// (C) Copyright 2019 Intel Corporation.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// GOVERNMENT LICENSE RIGHTS-OPEN SOURCE SOFTWARE
+// The Government's rights to use, modify, reproduce, release, perform, display,
+// or disclose this software are subject to the terms of the Apache License as
+// provided in Contract No. 8F-30005.
+// Any reproduction of computer software, computer software documentation, or
+// portions thereof marked with this legend must also reproduce the markings.
+//
+
+package security
+
+import (
+	"testing"
+)
+
+func TestCommonNameToComponent(t *testing.T) {
+	testCases := []struct {
+		testname   string
+		commonname string
+		expected   Component
+	}{
+		{"AdminCN", "admin", ComponentAdmin},
```

Comment:
Seems like it should test both "admin" and some value where admin is the prefix. "admindave" would be accepted by the current logic, for example.

Reply:
That is a good idea. I'll do both.

Reply:
Followed mike's suggestion below and will only respond to admin.